### DISCUSSION TITLE
Include Perfectionist like rules

### DISCUSSION THREAD

**Discussion by Goldziher** [astral-sh/ruff#15148] [Ideas]
There is a very nice eslint plugin called "eslint-plugin-perfectionist", which has an equally nice docs here: https://perfectionist.dev/

I would like to suggest including equivalent functionality in ruff. 

Obviously the import sorting is already taken over by isort, but the main benefit of this plugin is in the way it automatically arranges objects keys, destructuring assignments, invocations of builtins such as `String.includes` etc.

One place I always use IDE plugins to sort (alphabetically) keys is for explicit function kwargs in python.

E.g.
```python
def my_func(*, a: str, c: int, b: float) -> str:
  ...
```

I will manually ensure is:

```python
def my_func(*, a: str, b: float, c: int) -> str:
  ...
```

This is just a silly example, obviously there are many use cases that will benefit from being sorted automatically according to a scheme, to ensure codebase consistency and minimize nitpicking changes by enforcing conventions.

**Comment by MichaReiser** [comment]
Thanks for the suggestion.

I don't think we can add these rules to ruff today. They're too opinionated, without a clear value added. We may be able to add them once we finished #1774

**Reply by Goldziher** [reply]
> Thanks for the suggestion.
> 
> I don't think we can add these rules to ruff today. They're too opinionated, without a clear value added. We may be able to add them once we finished #1774

Opinionated if turned on 😉  - these can be off by default etc.

Furthermore, perfectionist has two sort settings - alphabetical and human-readable. The human-readable is according to line length, whereas the alphabetical is alphabetical. In my opinion the human-readable setting is rather subjective, whereas the alphabetical one (which i use), is a very common cultural convention - and as such is somewhat less subjective.
